src/nonstandard.jl

```diff
@@ -56,3 +56,18 @@ constructorof(::Type{<:LinRange}) = linrange_constructor
 ### Expr: args get splatted
 # ::Expr annotation is to make it type-stable on Julia 1.3-
 constructorof(::Type{<:Expr}) = (head, args) -> Expr(head, args...)::Expr
+
+### Cholesky
+setproperties(C::LinearAlgebra.Cholesky, patch::NamedTuple{()}) = C
+function setproperties(C::LinearAlgebra.Cholesky, patch::NamedTuple{(:L,)})
+    return LinearAlgebra.Cholesky(C.uplo === 'U' ? permutedims(patch.L) : patch.L, C.uplo, C.info)
+end
+function setproperties(C::LinearAlgebra.Cholesky, patch::NamedTuple{(:U,)})
+    return LinearAlgebra.Cholesky(C.uplo === 'L' ? permutedims(patch.U) : patch.U, C.uplo, C.info)
+end
+function setproperties(C::LinearAlgebra.Cholesky, patch::NamedTuple{(:UL,)})
+    return LinearAlgebra.Cholesky(patch.UL, C.uplo, C.info)
+end
+@nospecialize function setproperties(C::LinearAlgebra.Cholesky, patch::NamedTuple)
+    error("Can only patch one of :L, :U, :UL at the time")
```

Comment:
Maybe this could be made a
```suggestion
    throw(ArgumentError("Can only patch one of :L, :U, :UL at the time"))
```
I also wonder if the error message to cover cases where neither any of these options (or other combinations are specified) - or is this something that can't happen?

Reply:
Ah good point; I'll add that.